packages/widget/src/components/address-input/address-input.ts

```diff
@@ -0,0 +1,87 @@
+import { LitElement, html } from 'lit';
+import type { HTMLTemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { ethers } from 'ethers';
+import { Network } from '@buildwithsygma/sygma-sdk-core';
+import { when } from 'lit/directives/when.js';
+import { validateSubstrateAddress } from '../../utils';
+import { styles } from './styles';
+
+@customElement('sygma-address-input')
+export class AddressInput extends LitElement {
+  static styles = styles;
+  @property({
+    type: String
+  })
+  address: string = '';
+
+  @property({ attribute: false })
+  onHandleAddress?: (address: string) => void;
```

Comment:
maybe `onAddressChange` would make a bit more sense? also, not sure if this prop should be optional?

Reply:
shouldn't actually

Reply:
and what about changing name to something a bit more semantic?